File: Grocery_App/DataAccessLayer/Category.cs
using System.ComponentModel.DataAnnotations;

namespace DataAccessLayer
{
    public class Category
    {
        [Key]
        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        public string CategoryImage { get; set; }
        
        public ICollection<Products>? Products { get; }
    }
}

File: Angular_App/src/app/Models/category.ts
export class Category {
    categoryId:number=0;
    categoryName:string="";
    categoryImage?:string="";

    constructor(categoryId:number,categoryName:string,categoryImage:string){
        this.categoryId=categoryId;
        this.categoryName=categoryName;
        this.categoryImage=categoryImage;
    }
}
